selector/methods/distance.py

```diff
@@ -175,34 +209,31 @@ def select_from_cluster(self, X, size, labels=None):
         -------
         selected : list
             List of indices of selected samples.
-        """
-        if size > len(X):
-            raise ValueError(
-                f"Given size is greater than the number of sample points, {size} > {len(X)} "
-            )
 
+        """
         # calculate pairwise distance between points
-        X_dist = X
+        x_dist = x
         if self.fun_dist is not None:
-            X_dist = self.fun_dist(X)
-        # check X_dist is a square symmetric matrix
-        if X_dist.shape[0] != X_dist.shape[1]:
-            raise ValueError(f"The pairwise distance matrix must be square, got {X_dist.shape}.")
-        if np.max(abs(X_dist - X_dist.T)) > 1e-8:
+            x_dist = self.fun_dist(x)
+        # check x_dist is a square symmetric matrix
+        if x_dist.shape[0] != x_dist.shape[1]:
+            raise ValueError(f"The pairwise distance matrix must be square, got {x_dist.shape}.")
+        if np.max(abs(x_dist - x_dist.T)) > 1e-8:
             raise ValueError("The pairwise distance matrix must be symmetric.")
 
         if labels is not None:
             # extract pairwise distances from full pairwise distance matrix to obtain a new matrix
             # that only contains pairwise distances between samples within a given cluster.
-            X_dist = X_dist[labels][:, labels]
+            x_dist = x_dist[labels][:, labels]
 
-        # choosing initial point as the medoid (i.e., point with minimum cumulative pairwise
-        # distances to other points)
-        selected = [np.argmin(np.sum(X_dist, axis=0))]
+        # setting up initial point
+        selected = setup_reference_index(
+            x=None, x_dist=x_dist, ref_index=self.ref_index, fun_dist=None
+        )
```

Comment:
@FanwangM: use a more descriptive name, like `get_initial_selection`.

Reply:
Changed to `get_initial_selections`.